nuxeo-web-ui-ftest/test/features/step_definitions/recents.js

```diff
@@ -1,6 +1,28 @@
 'use strict';
 
 module.exports = function () {
-  this.Then('I can see the list of recently viewed documents',
-      () => this.ui.drawer.recents.waitForVisible().should.be.true);
+  this.Given(/^I have a clean recently viewed documents list/, () => {
+    fixtures.documents.clearLocalStorage(this.username, 'nuxeo-recent-documents');
+    fixtures.documents.reloadLocalStorage('* >>> nuxeo-recent-documents >>> #storage');
+  });
+
+  this.Given(/^I have document with path "(.+)" on the recently viewed documents list/, (path) => {
+    fixtures.documents.addToLocalStorage(path, this.username, 'nuxeo-recent-documents');
+  });
+
+  this.Then('I can see the list of recently viewed documents', () => {
+    this.ui.drawer.recents.waitForVisible().should.be.true;
+  });
+
+  this.Then('I can see the list of recently viewed documents has "$nb" items', (nb) => {
+    fixtures.documents.reloadLocalStorage('* >>> nuxeo-recent-documents >>> #storage');
```

Comment:
If you force the reload of the local storage, then it defeats the purpose of the test because we are obscuring the fact the local storage may not reload properly.

We should only force the reload of the local storage if we do put something inside through fixture i.e. by calling native api.
In this case, we want to test that the simple fact of navigating to a document will update the recent menu without explicitly forcing the `reload`

Reply:
Perhaps with the reload on the page helpers as suggested below we can call the reload method on the provision steps.